### PR TITLE
FIX: Removed old cases -> Added EXAT + new tests

### PR DESCRIPTION
**Pull request by apoorvyadav1111**
Hi @JyotinderSingh, I tried fixing the test case by adding type in the file but the test failed due to no support for EXAT option in set. I checked the PR for this addition and found out that subsequent PRs have removed the EXAT option. Hence, the EXAT support must be added again.
For now, I removed the test case testing EXAT functionality. 
Please review.

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Could you provide a link to the Commit that removed the support for this command?

**Comment by apoorvyadav1111** [comment]
Hi @JyotinderSingh, Please find the exact commit that removed the EXAT. https://github.com/DiceDB/dice/pull/218/commits/8cc98dc7a66ae3c2248313abaa7ef68eaeb036dd. 
It is from the same PR merge that pushed the test. Exat was implemented at the beginning and it seems like it was removed upon code review but the test cases where left intact. This lead to two issues, testing unimplemented code and referring to a type that was removed.

**Comment by soumya-codes** [comment]
@apoorvyadav1111 thanks for finding the issue. Instead of removing the test can we add back the code for EXAT option? wdyt?

**Comment by JyotinderSingh** [comment]
> @apoorvyadav1111 thanks for finding the issue. Instead of removing the test can we add back the code for EXAT option? wdyt?

Let's do this.

**Comment by apoorvyadav1111** [comment]
Hi @JyotinderSingh, @soumya-codes, Please review.

File: tests/set_test.go [hunk 1]
```diff
@@ -127,3 +127,18 @@ func TestSetWithOptions(t *testing.T) {
 		})
 	}
 }
+
+func TestSetWithExat(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+	Etime := strconv.FormatInt(time.Now().Unix()+10, 10)
+
+	assert.Equal(t, "OK", fireCommand(conn, "SET k v EXAT "+Etime), "Value mismatch for cmd SET k v EXAT "+Etime)
```
Comment (soumya-codes) [inline tests/set_test.go:136]: Add test for the negative condition where user can provide `EXAT` and `PXAT` in the same command.

**Review by soumya-codes — COMMENTED** [review]
Thanks for the quick follow-up. Have left couple of minor comments.

File: tests/set_test.go (hunk 1 above)
Comment (apoorvyadav1111) [reply tests/set_test.go]: Hi @soumya-codes, I have updated the code base as per your comments. Please review.

**Comment by apoorvyadav1111** [comment]
Hi @soumya-codes, @JyotinderSingh, addressed comments and combined PXAT EXAT. Please review

**Review by JyotinderSingh — APPROVED** [review]

**Review by soumya-codes — APPROVED** [review]
LGTM

File: tests/set_test.go [hunk 2]
```diff
@@ -127,3 +127,38 @@ func TestSetWithOptions(t *testing.T) {
 		})
 	}
 }
+
+func TestSetWithExat(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+	Etime := strconv.FormatInt(time.Now().Unix()+10, 10)
```
Comment (JyotinderSingh) [inline tests/set_test.go:134]: nit: We can likely use smaller expiry time (and accordingly update the test case) so that the overall test runs faster :)

Comment (apoorvyadav1111) [reply tests/set_test.go:134]: Sure. I can do it.

Comment (JyotinderSingh) [reply tests/set_test.go:134]: The PR looks good, just mentioned this. We can fix this separately.

Comment (apoorvyadav1111) [reply tests/set_test.go:134]: Ok, thanks for assigning this to me and sharing your views through it
